### ISSUE TITLE
Add language and filter buttons to the search bar on all pages

### ISSUE THREAD

**Issue by thesunshade** [suttacentral/suttacentral#2716]
#### User story
Someone is on a non-search result page and they want to search. They press the search button. Only after they submit a search can they see the language and filter icons.

E.g. non-search result page:

![image](https://github.com/suttacentral/suttacentral/assets/82448383/9d7c7794-f804-48c3-9ad0-474f5d1df898)

Only on a search result page can you see the buttons:

![image](https://github.com/suttacentral/suttacentral/assets/82448383/5e3635f5-2875-4fff-949f-d82b39ea3753)

#### Feature description

It could look something like this every time the search was opened:

![2023-07-09 15_06_42-_ Untitled -3 0 (RGB color 8-bit gamma integer, GIMP built-in sRGB, 2 layers) 11](https://github.com/suttacentral/suttacentral/assets/82448383/6b38c303-bbed-4c7e-b854-619ac50f9bf0)

However this could present problems on mobile because it reduces the amount of space left to type into the search bar.

#### Reason for the improvement

On a larger screen (any kind of non-mobile device), if the buttons are only available after the search results are given, they are out of the user's area of interest on the screen, so they may not see them as options:

![2023-07-07 19_02_31-Sutta Index — Firefox Developer Edition](https://github.com/suttacentral/suttacentral/assets/82448383/bc543a4f-663a-412c-991e-193a6b9c8e9f)

#### Pre milestone planning check:
- [ ] Small enough to completed in a milestone.
- [ ] Dependencies marked
- [ ] No external dependencies block the PBI from being completed.
- [ ] Details are understood by dev team to decide if the PBI can be completed.

#### Done check:
- [ ] Produced code for presumed functionalities
- [ ] Project builds without errors
- [ ] Peer Code Review performed/pull request approved x2
- [ ] Project deployed on the stage environment identical to production platform
- [ ] Feature is tested against acceptance criteria
- [ ] Feature ok-ed by Product Owner (moving to Closed on the Board)
- [ ] Refactoring completed
- [ ] Any configuration or build changes documented (readme, etc...)

**Comment by sujato** [comment]
Yes, you've identified the problem well, but I don't think this option will work on mobiles. Four icons is too much I think. Hmm, i wonder if there is another way. Maybe just show the simple interface on mobile? So they get the options after the search? Or, maybe we show the icons as you demonstrate, but hide them when they start typing?

Oh, another idea, maybe we don't need the X close icon? Usually on a mobile you get used to just tapping the screen somewhere else, they often don't have an explicit "dismiss" button. 

Just checking my stock android, if I click the default google search, it opens a field with three icons:

- `G` on the left, this takes you to the full search page
- Mic icon on the right for speaking
- "lens" icon for .. doing whatever lens does.

So yeah, three icons is probably about right. 

What do you think? 

- Four icons on larger screens
- Three icons on mobile, without the close button.

**Comment by thesunshade** [comment]
Different thought…

Does it really make sense that the language filters and the other filters are two different interfaces? Because they are all filters, eh? It seems a bit off that there are selectable options under both, but the filter section is more about providing documentation. 

The language options are so numerous that it's understandable they would be in their own section, but it seems a bit arbitrary. ¯\\\_(ツ)_/¯

BTW, and maybe this should be a different issue, (OK, new issue created: https://github.com/suttacentral/suttacentral/issues/2728)  but that language interface is broken on both Chrome and FF on the Android device I use. Even panning left and right and scrolling up and down, some things are hidden off screen. Root languages are not visible. The Filter section is also wonky because the example searches disappear off the right side. Seems like two columns won't work for that content.
